Packages/com.unity.inputsystem/InputSystem/Plugins/Android/AndroidSupport.cs

```diff
@@ -151,6 +154,19 @@ public static void Initialize()
                     .WithCapability("sensorType", AndroidSensorType.StepCounter));
 
             InputSystem.onFindLayoutForDevice += OnFindLayoutForDevice;
+
+            #if !UNITY_EDITOR
+            // AndroidSupport.Initialize can be called multiple times when running tests
+            // Don't recreate keyboard
+            NativeInputRuntime.instance.screenKeyboard = AndroidScreenKeyboard.instance;
+            #endif
+        }
+
+        public static void Shutdown()
+        {
+            #if !UNITY_EDITOR
+            NativeInputRuntime.instance.screenKeyboard = null;
```

Comment:
We should no only null the instance, but also dispose AndroidJavaObject/Proxy stuff.

Reply:
Added disposing both for iOS and Android